validator/client/src/main/java/tech/pegasys/teku/validator/client/AbstractDutyLoader.java

```diff
@@ -45,15 +45,19 @@ protected AbstractDutyLoader(
   public SafeFuture<ScheduledDuties> loadDutiesForEpoch(final UInt64 epoch) {
     LOG.trace("Requesting attestation duties for epoch {}", epoch);
     final ScheduledDuties scheduledDuties = scheduledDutiesFactory.get();
-    return requestDuties(epoch, validatorIndexProvider.getValidatorIndices(validators.keySet()))
-        .thenApply(
-            maybeDuties ->
-                maybeDuties.orElseThrow(
-                    () ->
-                        new NodeDataUnavailableException(
-                            "Duties could not be calculated because chain data was not yet available")))
-        .thenCompose(duties -> scheduleAllDuties(scheduledDuties, duties))
-        .thenApply(__ -> scheduledDuties);
+    return validatorIndexProvider
+        .getValidatorIndices(validators.keySet())
+        .thenCompose(
+            validatorIndices ->
+                requestDuties(epoch, validatorIndices)
+                    .thenApply(
```

Comment:
nit: you can reduce the level of nesting here slightly with:
```suggestion
        .thenCompose(validatorIndices -> requestDuties(epoch, validatorIndices))
        .thenApply(
```

ie `requestDuties` is in its own `thenCompose`